src/EditorFeatures/CSharpTest/Diagnostics/UseImplicitOrExplicitType/UseImplicitTypeTests.cs

```diff
@@ -1441,5 +1442,24 @@ static void M()
                 diagnosticId: IDEDiagnosticIds.UseImplicitTypeDiagnosticId,
                 diagnosticSeverity: DiagnosticSeverity.Error);
         }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CodeActionsUseImplicitType), Trait(Traits.Feature, Test.Utilities.Traits.Features.Tuples)]
+        public async Task SuggestVarOnLocalWithIntrinsicTypeTuple()
+        {
+            await TestAsync(
+@"class C { static void M() { [|(int a, string)|] s = (a: 1, ""hello""); } }",
+@"class C { static void M() { var s = (a: 1, ""hello""); } }", 
+options: ImplicitTypeEverywhere(),
+parseOptions: TestOptions.Regular.WithTuplesFeature());
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CodeActionsUseImplicitType), Trait(Traits.Feature, Test.Utilities.Traits.Features.Tuples)]
+        public async Task TupleWithDifferentNames()
+        {
+            await TestMissingAsync(
+@"class C { static void M() { [|(int, string)|] s = (c: 1, d: ""hello""); } }",
+options: ImplicitTypeEverywhere(),
+parseOptions: TestOptions.Regular.WithTuplesFeature());
+        }
```

Comment:
I added a test for `ValueTuple<...> x = ValueTuple.Create(...);`. It works as expected (the type is apparent from the right-hand side).

But for `(int a, bool) x = (a: 1, true);` the type is not apparent. Filed issue #11094.

For `(int, int) x = ValueTuple.Create(1, 1);`, this also won't work yet. But I think it will work later this week due to some pending compiler changes. Filed issue #11095.